src/components/user-defined-sets/NamedSetManager.js

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable jsx-a11y/accessible-emoji */
+import React, { useState } from 'react';
+import * as Sets from './sets';
+
+export default function NamedSetManager(props) {
+  const {
+    sets,
+    name,
+    onUpdateSets = (msg) => {
+      console.warn(`onUpdateSets from NamedSetManager ${msg}`);
+    },
+  } = props;
+  const set = sets.namedSets[name];
+  const [isEditing, setIsEditing] = useState(false);
+  const [setName, setSetName] = useState(name);
+
+  const renameSet = () => {
+    setIsEditing(false);
+    onUpdateSets(Sets.renameNamedSet(sets, name, setName));
+  };
+
+  const deleteSet = () => {
+    setIsEditing(false);
+    onUpdateSets(Sets.deleteNamedSet(sets, name));
+  };
+
+  const updateCurrentSet = (nameToSelect) => {
+    onUpdateSets(Sets.setCurrentSet(sets, sets.namedSets[nameToSelect]));
+  };
+
+  return (
+    <React.Fragment>
+      <tr className="set-name">
+        <td>
+          {isEditing
+            ? (
+              <input
+                value={setName}
+                onKeyDown={e => (e.keyCode === 13 ? renameSet() : null)}
+                onChange={e => setSetName(e.target.value)}
+                type="text"
+              />
+            )
+            : <button type="button" className="select-button" onClick={() => updateCurrentSet(name)}>{name}</button>}
+        </td>
+        {isEditing
+          ? (
+            <td className="set-edit">
+              <button type="button" onClick={deleteSet}>🗑</button>
+            </td>
+          )
+          : (
+            <td className="set-count">
+              <small>{set.length}</small>
+            </td>
+          )}
+        {isEditing
+          ? (
+            <td className="set-edit">
+              <button type="button" onClick={renameSet}>💾</button>
+            </td>
+          )
+          : (
+            <td className="set-edit">
+              <button type="button" className="edit-button" onClick={() => setIsEditing(true)}>✏️</button>
+            </td>
+          )
+        }
+      </tr>
```

Comment:
There's really very little that's shared outside the if-thens... Would it be more clear just to have one if-then at the top? You might even pull it all the way out of the JSX:
```
if (isEditing) {
 return <tr>...</tr>
}
return <tr>...</tr>
```